main/src/ca/uwaterloo/flix/language/phase/unification/BoolFormulaSubstitution.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ *  Copyright 2020 Magnus Madsen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package ca.uwaterloo.flix.language.phase.unification
+
+import ca.uwaterloo.flix.language.ast.Symbol
+import ca.uwaterloo.flix.util.InternalCompilerException
+import ca.uwaterloo.flix.util.collection.Bimap
+
+/**
+  * Companion object for the [[BoolFormulaSubstitution]] class.
+  */
+object BoolFormulaSubstitution {
+  /**
+    * Returns the empty substitution.
+    */
+  def empty[F]: BoolFormulaSubstitution[F] = BoolFormulaSubstitution(Map.empty)
```

Comment:
val, for performance.

Reply:
Can you do a polymorphic val?

Reply:
Like Nil?

Reply:
Scala doesn't like it :D

Reply:
Hmm okay then :)

Reply:
val empty: BoolFormula[Nothing] = ... ??